openbb_platform/providers/polygon/openbb_polygon/models/market_snapshots.py

```diff
@@ -0,0 +1,246 @@
+    """Polygon Market Snapshots Data."""
+
+    vwap: Optional[float] = Field(
+        description="The volume weighted average price of the stock on the current trading day.",
+        default=None,
+    )
+    prev_open: Optional[float] = Field(
+        description="The previous trading session opening price.", default=None
+    )
+    prev_high: Optional[float] = Field(
+        description="The previous trading session high price.", default=None
+    )
+    prev_low: Optional[float] = Field(
+        description="The previous trading session low price.", default=None
+    )
+    prev_volume: Optional[float] = Field(
+        description="The previous trading session volume.", default=None
+    )
+    prev_vwap: Optional[float] = Field(
+        description="The previous trading session VWAP.", default=None
+    )
+    last_updated: Optional[datetime] = Field(
+        description="The last time the data was updated."
+    )
+    bid: Optional[float] = Field(description="The current bid price.", default=None)
+    bid_size: Optional[int] = Field(description="The current bid size.", default=None)
+    ask_size: Optional[int] = Field(description="The current ask size.", default=None)
+    ask: Optional[float] = Field(description="The current ask price.", default=None)
+    quote_timestamp: Optional[datetime] = Field(
+        description="The timestamp of the last quote.", default=None
+    )
+    last_trade_price: Optional[float] = Field(
+        description="The last trade price.", default=None
+    )
+    last_trade_size: Optional[int] = Field(
+        description="The last trade size.", default=None
+    )
+    last_trade_conditions: Optional[List[int]] = Field(
+        description="The last trade condition codes.", default=None
+    )
+    last_trade_exchange: Optional[int] = Field(
+        description="The last trade exchange ID code.", default=None
+    )
+    last_trade_timestamp: Optional[datetime] = Field(
+        description="The last trade timestamp.", default=None
+    )
+
+
+class PolygonMarketSnapshotsFetcher(
+    Fetcher[
+        PolygonMarketSnapshotsQueryParams,
+        List[PolygonMarketSnapshotsData],
+    ]
+):
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> PolygonMarketSnapshotsQueryParams:
+        return PolygonMarketSnapshotsQueryParams(**params)
+
+    @staticmethod
+    def extract_data(
+        query: PolygonMarketSnapshotsQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        api_key = credentials.get("polygon_api_key") if credentials else ""
+
+        url = f"https://api.polygon.io/v2/snapshot/locale/us/markets/stocks/tickers?apiKey={api_key}"
+
+        response = {}
+        r = make_request(url)
+        if r.status_code == 200 and r.json()["status"] == "OK":
+            response = r.json()["tickers"]
+
+        results = pd.DataFrame()
+
+        symbol: List[str] = []
```

Comment:
The code below is very smelly. This has to be shortened to reduce the clutter and improve readability. Some part of it has to be done in the `extract_data` function.

Reply:
wdym? This is the `extract_data` function.  If things are not renamed it comes out with triplicates of `o`, `h`, `l`, `c`, `v`, etc.